apps/editing-toolkit/editing-toolkit-plugin/error-reporting/index.php

```diff
@@ -0,0 +1,134 @@
+ */
+function activate_error_reporting() {
+	add_action( 'admin_print_scripts', __NAMESPACE__ . '\head_error_handler' );
+	add_filter( 'script_loader_tag', __NAMESPACE__ . '\add_crossorigin_to_script_els', 99, 2 );
+	// We load as last as possible for perf reasons. The head handler will
+	// capture errors until the main handler is loaded.
+	add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\enqueue_script', 99 );
+}
+
+/**
+ * Temporary function to feature flag by segment. We'll be gradually testing
+ * it on production simple sites, observing how it works and and gradually
+ * increase the segment until we reach 100 and eventually just remove this logic.
+ *
+ * It's important to note that we'll only gradually roll-out for simple sites.
+ * Once we're confident enough it works well on simple-sites, we'll work on
+ * addressing issues that are preventing this module from working on WoA, test
+ * on real AT sites and finally deploy the plugin to WoA while removing the
+ * guard above that prevents it from loading on WoA. The reason for that in WoA
+ * it's actually trivial to test a custom build in production without affecting
+ * other customers, and on simple sites, it's not, hence the need for a gradual
+ * roll-out as an additional safety measure.
+ *
+ * @todo Remove once the roll-out is complete.
+ *
+ * @param int $user_id the user id.
+ * @return bool
+ */
+function user_in_test_segment( $user_id ) {
+	$current_segment = 10; // segment of existing users that will get this feature.
+	$user_id         = get_current_user_id();
+	$user_segment    = $user_id % 100;
+
+	// We get the last two digits of the user id and that will be used to decide in what
+	// segment the user is. i.e if current_segment is 10, then only ids that end in < 10
+	// will be considered part of the segment.
+	return $user_segment < $current_segment;
+}
+
+/**
+ * Returns whether or not the user is an automattician, but first verifies
+ * if the function `is_automattician` exists in the current context. If not,
+ * then we consider the user to not be an automattician. This guard is needed
+ * because this function is not present in some envs, namely the testing env.
+ *
+ * @todo Remove once roll-out is complete
+ *
+ * @param int $user_id the user id.
+ * @return bool
+ */
+function user_is_automattician( $user_id ) {
+	return function_exists( 'is_automattician' ) && is_automattician( $user_id );
+}
+
+/**
+ * Returns whether or not the site loading ETK is in the WoA env.
+ *
+ * @return bool
+ */
+function is_atomic() {
+	return defined( 'IS_ATOMIC' ) && IS_ATOMIC;
+}
+
+// We don't want to activate this module in AT just yet. See https://wp.me/p4TIVU-9DI#comment-10922.
+// @todo Remove once we have a version that works for WPCOM simple sites and WoA.
+if ( ! is_atomic() ) {
+	// @todo Remove once the roll-out is complete.
+	$user_id = get_current_user_id();
+	if ( user_is_automattician( $user_id ) || user_in_test_segment( $user_id ) ) {
```

Comment:
Is `user_in_test_segment` present in all the environments we care about?

Otherwise looking pretty good!

Reply:
It's defined in the same file: https://github.com/Automattic/wp-calypso/pull/49485/files/8dc23aedb7e11d6c92c8725b12288aa43bdd1334#diff-75804e0a48800cac5f55add252a9f74d7004a83d23084828643cd76e7f724f9bR91. So yeah, it should!

Reply:
:man_facepalming: My bad! Thought it was a wpcom thing

Reply:
No problem at all!